app/forms/coupon_form.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class CouponForm < ApplicationForm
+  attr_accessor :order_id, :code
+
+  validate :unique_order
+
+  def save
+    return unless valid?
+
+    @model.update(order_id: @params[:order_id])
+    @model
+  end
+
+  private
+
+  def unique_order
+    errors.add(:order_id) if @model.order
```

Comment:
what message?

Reply:
if coupon_form.save
      show_good_flash
else
     here i show message for uniqe order
end
and i create error for nil from save
But now, i fixed to 
if coupon_form.save
      redirect_with_flash(notice: t('carts.message.valid_coupon'))
    else
      redirect_with_flash(alert: coupon_form.errors[:order_id].to_sentence)
    end, also add message where u commit